chirp/drivers/baofeng_uv17.py

```diff
@@ -30,11 +30,25 @@
 LIST_SCANMODE = ["Time", "Carrier", "Search"]
 
 
+# Radios seem to have different memory sizes
+def _get_memory_size(radio):
+    response = baofeng_uv17Pro._sendmagic(radio, radio._magic_memsize[0][0],
+                                          radio._magic_memsize[0][1])
+    mem_size = struct.unpack("<I", response[7:])[0]
+    for magic, resplen in radio._magics2:
+        baofeng_uv17Pro._sendmagic(radio, magic, resplen)
+    return mem_size
+
+
 # Locations of memory may differ each time, so a mapping has to be made first
 def _get_memory_map(radio):
     # Get memory map
     memory_map = []
-    for addr in range(0x1FFF, 0x10FFF, 0x1000):
+    mem_size = _get_memory_size(radio)
+    if mem_size != radio._radio_memsize:
+        raise errors.RadioError('Memory size doesn\'t match ' +
+                                'the selected radio.')
```

Comment:
It would probably be better to make this something a little more obvious. People will open tickets saying "I got an error from the radio" and we'll close it with "You're using the wrong model". So making this more direct will help.

Since I need to rebase this and fix your commit message title, I'll edit this too, if you don't mind :)